client/src/send_and_confirm_transactions_in_parallel.rs

```diff
@@ -0,0 +1,278 @@
+                    let transactions_to_verify: Vec<Signature> = transaction_map
+                        .iter()
+                        .filter(|x| {
+                            current_block_height < x.blockheight + MAX_RECENT_BLOCKHASHES as u64
+                        })
+                        .map(|x| *x.key())
+                        .collect();
+                    for signatures in
+                        transactions_to_verify.chunks(MAX_GET_SIGNATURE_STATUSES_QUERY_ITEMS)
+                    {
+                        if let Ok(result) = rpc_client.get_signature_statuses(signatures).await {
+                            let statuses = result.value;
+                            for (signature, status) in signatures.iter().zip(statuses.into_iter()) {
+                                if let Some(status) = status {
+                                    if status.satisfies_commitment(rpc_client.commitment()) {
+                                        if let Some((_, data)) = transaction_map.remove(signature) {
+                                            confirmed_transactions.fetch_add(1, Ordering::Relaxed);
+                                            let mut transaction_errors =
+                                                transaction_errors.lock().await;
+                                            transaction_errors[data.index] = status.err;
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+                tokio::time::sleep(CONFIRMATION_REFRESH_RATE).await;
+            }
+        });
+    }
+
+    // transaction sender task
+    let total_transactions = messages.len();
+    let mut initial = true;
+    for expired_blockhash_retries in (0..5).rev() {
+        // only send messages which have not been confirmed
+        let nb_transaction = messages.len();
+        let messages: Vec<(usize, Message)> = if initial {
+            initial = false;
+            messages.iter().cloned().enumerate().collect()
+        } else {
+            // remove all the confirmed transactions
+            transaction_map
+                .iter()
+                .map(|x| (x.index, x.transaction.message.clone()))
+                .collect()
+        };
+
+        if messages.is_empty() {
+            break;
+        }
+
+        // clear the map so that we can start resending
+        transaction_map.clear();
+
+        for (index, message) in messages.iter() {
+            let mut transaction = Transaction::new_unsigned(message.clone());
+            let blockhash = *block_hash_rw.read().await;
+            transaction.try_sign(signers, blockhash)?;
+            let signautre = transaction.signatures[0];
+            if !tpu_client.send_transaction(&transaction).await {
+                let _ = tpu_client
```

Comment:
why's this return ignored?  there's no reason to track a tx that failed bcast

Reply:
This is how it is handled previously https://github.com/solana-labs/solana/blob/63baab57fc324937dc1d72b288ba55279c8e793e/tpu-client/src/nonblocking/tpu_client.rs#L487C23-L487C23 .

I think it is correctly handled because the TPU probably received the transaction but there was error while finishing the connection. That is why TPU Client `send_transaction` failed, so for safe side we sent it to RPC client.